src/propertySelector/propertySelector.js

```diff
@@ -0,0 +1,150 @@
+    let search = '';
+
+    const parentGap = 20;
+
+    /**
+     * Lookup for characters in text to highlight
+     * @param {String} text - text to lookup
+     * @param {String} highlight - character(s) to be highlighted
+     * @param {regExp|String} match - match to be applied in the text
+     * @returns {String} - highlighted text
+     */
+    function highlightCharacter(text, highlight, match) {
+        return text.replace(match, `<b>${highlight}</b>`);
+    }
+
+    /**
+     * Creates property description list element
+     * @param {*} property
+     * @returns JQuery element containing property description
+     */
+    function createPropertyOption(property) {
+        const descriptionData = Object.assign({}, property);
+        if (search !== '') {
+            descriptionData.label = highlightCharacter(descriptionData.label, search, search);
+        }
+        const $propertyDescription = $(propertyDescriptionTpl({ property: descriptionData }));
+        const $checkboxContainer = $('.checkbox-container', $propertyDescription);
+        $checkboxContainer.append(checkBoxTpl({ id: descriptionData.id, checked: descriptionData.selected }));
+        const $checkbox = $('input', $checkboxContainer);
+        $checkbox.on('change', function () {
+            if (this.checked) {
+                selectedProperties.add(property.id);
+            } else {
+                selectedProperties.delete(property.id);
+            }
+        });
+        return $propertyDescription;
+    }
+
+    const instance = component({
+        positionContainer: function positionContainer() {
+            let { top, left, right, bottom } = this.config.data.position;
+            let maxHeight;
+            if (typeof bottom === 'undefined') {
+                maxHeight = $container.parent().height() - top - parentGap;
+            }
+            if (typeof top === 'undefined') {
+                maxHeight = $container.parent().height() - bottom - parentGap;
+            }
+            $container.css({ top, left, right, bottom, maxHeight });
+        },
+
+        /**
+         * Updates the list
+         */
+        redrawList: function redrawList() {
+            $propertyListContaner.empty();
+            availableProperties.forEach(property => {
+                property.selected = selectedProperties.has(property.id);
+                if (search === '' || property.label.toLowerCase().includes(search.toLowerCase())) {
+                    $propertyListContaner.append(createPropertyOption(property));
+                }
+            });
+        },
+        /**
+         * Adds and setups buttons to button container
+         */
+        addButtons: function addButtons() {
+            const cancelButton = buttonFactory({
+                id: 'cancel',
+                label: 'Cancel',
+                type: 'info',
+                cls: 'btn-secondary'
+            }).on('click', () => {
+                this.trigger('cancel');
```

Comment:
maybe cancel also should clear selected options?

Reply:
**chore:** Actually, the component should close when either the `cancel` or the `save` button is clicked.

There are several ways of doing it. Either the component self-destroys or the component hides. 
The first solution means the consumer needs to recreate the component for each use.
The second solution means the component must implement an API to open in demand. with a possible new selection.

Reply:
Nope, It should not affect the selected props set